src/dscanner/analysis/assert_without_msg.d

```diff
@@ -5,132 +5,165 @@
 module dscanner.analysis.assert_without_msg;
 
 import dscanner.analysis.base;
-import dscanner.utils : safeAccess;
-import dsymbol.scope_ : Scope;
-import dparse.lexer;
-import dparse.ast;
-
 import std.stdio;
-import std.algorithm;
 
 /**
  * Check that all asserts have an explanatory message.
  */
-final class AssertWithoutMessageCheck : BaseAnalyzer
+extern(C++) class AssertWithoutMessageCheck(AST) : BaseAnalyzerDmd!AST
 {
-	enum string KEY = "dscanner.style.assert_without_msg";
-	enum string MESSAGE = "An assert should have an explanatory message";
 	mixin AnalyzerInfo!"assert_without_msg";
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
 	///
-	this(string fileName, const(Scope)* sc, bool skipTests = false)
+	extern(D) this(string fileName, bool skipTests = false)
 	{
-		super(fileName, sc, skipTests);
+		super(fileName, skipTests);
+	
+		import std.file : readText;
+		this.code = readText(fileName);
 	}
 
-	override void visit(const AssertExpression expr)
+	mixin ScopedVisit!(AST.FuncDeclaration);
+	mixin ScopedVisit!(AST.IfStatement);
+	mixin ScopedVisit!(AST.WhileStatement);
+	mixin ScopedVisit!(AST.ForStatement);
+	mixin ScopedVisit!(AST.ForeachStatement);
+	mixin ScopedVisit!(AST.ClassDeclaration);
+	mixin ScopedVisit!(AST.StructDeclaration);
+	mixin ScopedVisit!(AST.InterfaceDeclaration);
+	mixin ScopedVisit!(AST.UnionDeclaration);
+	mixin ScopedVisit!(AST.ScopeStatement);
+
+	override void visit(AST.AssertExp ae)
 	{
-		if (expr.assertArguments && expr.assertArguments.message is null)
-			addErrorMessage(expr.line, expr.column, KEY, MESSAGE);
+		// We need to ensure that this is an actual `assert` expression,
+		// Not an in/out contract represented also as an AssertExp in the AST
+		import std.string : indexOf;
+
+		// File Offset indicates where the expression starts
+		// To extract the function name get rid of `(` and everything that follows
+		string funcName = code[ae.loc.fileOffset .. $];
+		funcName = funcName[0 .. funcName.indexOf("(")];
```

Comment:
This is wrong and may be the cause for your memory corruption. `in` and `out` contracts are kept in [func.d](https://github.com/dlang/dmd/blob/master/compiler/src/dmd/func.d#L258). You can just skip analyzing those when traversing the AST. Take a look at the [transitive visitor](https://github.com/dlang/dmd/blob/master/compiler/src/dmd/transitivevisitor.d#L835) to make sure you are correctly implementing this.